staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -265,38 +303,25 @@ func (r *crdHandler) removeDeadStorage() {
 		if !found {
 			glog.V(4).Infof("Removing dead CRD storage for %v", s.requestScope.Resource)
 			s.storage.DestroyFunc()
```

Comment:
This storage could still accept new requests until storageMap2 is stored, and could still be handling previously accepted requests even after that. @sttts, what does this destroy func do, and does it gracefully shut down?

Reply:
Note this is preexisting, but wanted to point it out

Reply:
> This storage could still accept new requests until storageMap2 is stored

Then we should defer that call. Doing that a moment later does not hurt as long as it is called.

It stops the etcd watch cache logic and some metrics related code (we leaked lots of memory in 1.8 without this).